src/StrawberryfieldFilePersisterService.php

```diff
@@ -824,7 +822,46 @@ public function removefromAsFileStructure(
           $originaljson[self::AS_TYPE_PREFIX . $as_file_type][self::FILE_IRI_PREFIX . $uuid]['dr:fid'] == $file->id(
           )) {
           unset($originaljson[self::AS_TYPE_PREFIX . $as_file_type][self::FILE_IRI_PREFIX . $uuid]);
-          $this->remove_file_usage($file, $nodeid, 'node', 1);
+          // We can only remove usage for already saved content entities.
+          if (!$entity->isNew()) {
+            $this->remove_file_usage($file, $entity->id(), 'node', 1);
+          }
+        }
+      }
+    }
+
+    $not_existing = array_diff($file_id_list, $existing_ids);
+
+    // means we have left over files (coming from dr:fid that DO not longer
+    // exist inside Drupal as entities. Clean this up but in a costly way.
+    if (count($not_existing) > 0) {
+      $originaljson = $this->removefromAsFileStructureBrutForce($not_existing, $originaljson);
+    }
+
+    return $originaljson;
+  }
+
+  /**
+   * Removes a list of File IDs from the as:structure in a non optimal way.
+   *
+   * @param array $file_id_list
+   * @param array $originaljson
+   *
+   * @return array
+   */
+  public function removefromAsFileStructureBrutForce(
```

Comment:
I *think* if an method param is nullable, it should be at the end of the list

Reply:
In this case its really not nullable, since it needs to be an array, even if empty. For this function and its more precise counterpart I removed the defaults at all. Mostly because I did not wanted to change the signature's order. Thanks. Hope that addresses the problem

Reply:
It does, thanks!